src/nvim/terminal.c

```diff
@@ -644,6 +653,57 @@ Buffer terminal_buf(const Terminal *term)
 // }}}
 // libvterm callbacks {{{
 
+#define STATIC_CSTR_AS_STRING(s) ((String) {.data = s, .size = sizeof(s) - 1})
+
+static int term_unhandled_csi(const char *leader, const long args[], int argcount,
+                              const char *intermed, char command, void *user)
+{
+  Array numbers = ARRAY_DICT_INIT;
+  Array arguments = ARRAY_DICT_INIT;
+  Error err = ERROR_INIT;
+
+  for (int i = 0; i < argcount; i++) {
+    ADD(numbers, INTEGER_OBJ(args[i]));
+  }
+
+  ADD(arguments, ARRAY_OBJ(numbers));
+  ADD(arguments, STRING_OBJ(STATIC_CSTR_AS_STRING(";")));
+
+  Object rv = nvim_execute_lua(STATIC_CSTR_AS_STRING("return table.concat(...)"),
```

Comment:
The standard pattern for "building up" strings in C code is otherwise to use growarray, i e
```
garray_T ga;
ga_init(&ga, (int)sizeof(char *), 10);

// in a loop or something
snprintf((char *)IObuff, IOSIZE, "%d", idx);
ga_concat(&ga, IObuff);

ga_append(&ga, ';');
```

perhaps we should have `ga_printf` directly also, so it is easier to use.